# bioinformatics KU course
A repository for the crash-course on microbial (meta)genomics data analysis at the University of Copenhagen, <a href="https://food.ku.dk/english/research_at_food/sections/microbiology/">Food Microbiology, Gut Health, and Fermentation</a>.

<img  >
<figure>
    <img src="https://github.com/farhadm1990/bioinformatics_KU/blob/main/pix/logo.png" alt="Description of the image" width="400">
    <figcaption>This is image was generated by Dall.E</figcaption>
</figure>

## This course will cover two main tools:

# <a href="">1. MicroLoop</a>:

<p style="color: purple">Microbiome data analysis has posed many challenges on the researchers and decision-making entities, due to its complexity and hefty work that comes with it. In addition, due to lack of a golden standard for this type of sparce data analysis reproducibility of the results from lab to lab might be jeopardized. Therefore, MicroLoop was developed as a shiny app in R to perfom a wide variety of task with no need for hard-coding and any demanding bioinformatics skills.

<h5>The app offers modules for the following analysis:</h5>
<ol>
<li><h5>Reading a variety of data formats</h5></li>
<li><h5>Basic inferences on the data</h5></li>
<li><h5>Preprocessing and cleaning up the data</h5></li>
<li><h5>Mixed-model complex statistical analysis on within sample variance <strong>(Alpha diversity)</strong></h5></li>
<li><h5>Data normalization and multivariate analysis of variance between groups <strong>(Beta diversity)</strong></h5></li>
<li><h5>Creating stacked bar plot on the distribution of different taxa across different variables (groups/samples)</h5></li>
<li><h5>Differential abundance analysis with multiple known tools</h5></li>
<li><h5>Association/correlation analysis between numerical variables and mcirbtiome data</h5></li>
<li><h5>Phylogenetic-based analysis and tree visualizations</h5></li>
</ol>

</p>

<h6>This app is hosted on an <a href="https://aws.amazon.com/ec2/"> <strong>ECS</strong></a> inctance on <a href="https://aws.amazon.com/?nc2=h_lg">Amazon Web Service (AWS) </a> </h6>


# <a href="https://github.com/farhadm1990/bactflow">2. BactFlow</a>:

<p>BactFlow is a workflow for bacterial genome assembly of single isolate and metagenomics sequencing reads extracted from Oxford Nanopore Technology (ONT) and Illumina platforms. It is designed using Nextflow DSL 2 technology and reads the generic outputs of Guppy and Dorado basecallers.
This app is now available as with an interactive user interface (UI) and comes in three single-page apps:

<ul>
<li><h5>Pre-assembly:</h5></li>
<li><h5>Assembly:</h5></li>
<li><h5>Post-assembly:</h5></li>
</ul>
</p>

